data-prepper-plugins/otel-proto-common/src/test/java/org/opensearch/dataprepper/plugins/otel/codec/OTelLogsProtoBufDecoderTest.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.otel.codec;
+
+import java.io.InputStream;
+import java.time.Instant;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+import org.opensearch.dataprepper.model.log.OpenTelemetryLog;
+import org.opensearch.dataprepper.model.event.Event;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+
+public class OTelLogsProtoBufDecoderTest {
+    private static final String TEST_REQUEST_LOGS_FILE = "test-otel-log.protobuf";
```

Comment:
Using binary data in tests is difficult to maintain. I wouldn't even know how to debug or update this test.

A better approach would be to dynamically generate the binary data in a `@BeforeEach`. You can generate it, put it into a `byte[]` and then create an ByteArrayInputStream instead of reading the resource.

Reply:
Added comments explaining how the binary data is generated.

Reply:
We have been putting "Developer Guide" sections in some README.md files when there is something complicated like this. Perhaps add some additional guidance on this in the README.md.